src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php

```diff
@@ -430,6 +433,10 @@ private function getTagVersions(array $tagsByKey, array &$invalidatedTags = [])
             if (isset($invalidatedTags[$tag])) {
                 $invalidatedTags[$tag] = $version->set(++$tagVersions[$tag]);
             }
+            if (!\is_int($tagVersions[$tag])) {
```

Comment:
on the line above, we have a `++` on this entry, shouldn't we check for this earlier?

Reply:
++ operator works with strings
Although it not always changes string values (and thus no invalidation happens in those cases) but at least it doesn't do the fatality.